happy/utils/IP.py

```diff
@@ -162,6 +168,34 @@ def __ipv6PrefixMatchAddress(prefix, mask, addr):
         return prefix[:plen] == addr[:plen]
 
 
+    @staticmethod
+    def __ipv4Canonicalize(addr):
+        """This function attempts to canonicalize an IPv4 address (four completely-specified octets) or prefix (less than four octets) into a four octet, dotted decimal string representation by appending the appropriate number of zeroes (.0) to the address. Note, an empty string is canonicalized into the any address (0.0.0.0).
+
+        Args:
+           addr (str): A string containing the IPv4 address or prefix to canonicalize
+
+        Returns:
+           str. The canonicalized IPv4 address or prefix.
+        """
+        
+        if len(addr) > 0:
+            octets = addr.count(".") + 1
+        else:
+            octets = 0
+
+        missing = 4 - octets
+
+        if missing > 0:
+            padding = ".".join(["0"] * missing)
```

Comment:
do you need ")"?
I am thinking to introduce unit test in happy, so that we can show proven of the code tested.